### PR TITLE
BugFix: updateLabel can updated not allowed fields

### LINKED ISSUE #11238: [bug] [api] issueGetLabels dont return correct responce
Gitea Version: 1.12.0+dev-243-g63c7b4e63

issueGetLabels (`PUT /repos/{owner}/{repo}/issues/{index}/labels`) return lables with empty propertys

get:
```json
[
  {
    "id": 312,
    "name": "",
    "color": "",
    "description": "",
    "url": ""
  }
]
```

expect:
```json
[
  {
    "id": 312,
    "name": "blau",
    "color": "0052cc",
    "description": "",
    "url": ""
  }
]
```

### PR DESCRIPTION
**Pull request by 6543**
fix #11238

### CODE REVIEW

File: models/issue_label.go [hunk 1]
```diff
@@ -587,7 +587,16 @@ func GetLabelsByIssueID(issueID int64) ([]*Label, error) {
 	return getLabelsByIssueID(x, issueID)
 }
 
-func updateLabel(e Engine, l *Label) error {
+func updateLabelProperties(e Engine, l *Label) error {
+	return updateLabel(e, l, "name", "description", "color")
+}
+
+func updateLabelNum(e Engine, l *Label) error {
```
Comment (lafriks) [inline models/issue_label.go]: There is no point for this function

**Review by lafriks — CHANGES_REQUESTED** [review]

**Review by zeripath — APPROVED** [review]

File: models/issue_label.go [hunk 2]
```diff
@@ -246,7 +246,7 @@ func UpdateLabel(l *Label) error {
 	if !LabelColorPattern.MatchString(l.Color) {
 		return fmt.Errorf("bad color code: %s", l.Color)
 	}
-	return updateLabel(x, l)
+	return updateLabelCols(x, l, "name", "description", "color", "num_issues", "num_closed_issues")
```
Comment (lafriks) [inline models/issue_label.go]:
```suggestion
	return updateLabelCols(x, l, "name", "description", "color")
```

**Review by lafriks — APPROVED** [review]

**Comment by 6543** [comment]
ping lgtm bot

**Comment by guillep2k** [comment]
@6543 I marked this as skip-changelog as I believe the bug was introduced in the 1.12-dev branch. Please correct me if I'm wrong.

**Comment by 6543** [comment]
you are right ... :)

**Pull request merged**